skel/handlers/middlewares.go

```diff
@@ -12,3 +17,82 @@ func VaultConfigMiddleware(conf *sonic.VaultConfig) gin.HandlerFunc {
 		c.Next()
 	}
 }
+
+// OpendaxConfigMiddleware middleware to set kaigara config to gin context
+func OpendaxConfigMiddleware(config *sonic.OpendaxConfig) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		c.Set("OpendaxConfig", config)
+		c.Next()
+	}
+}
+
+// GlobalVaultServiceMiddleware middleware to set global vault service to gin context
+func GlobalVaultServiceMiddleware(vaultService *vault.Service) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		c.Set("GlobalVaultService", vaultService)
+		c.Next()
+	}
+}
+
+// AuthMiddleware middleware to verify bearer token
+func AuthMiddleware() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		// Get bearer token from header
+		authHeader := strings.Split(c.GetHeader("Authorization"), "Bearer ")
+		if len(authHeader) != 2 {
+			c.Abort()
+			c.JSON(http.StatusBadRequest, gin.H{"error": "Authorization header not found"})
+			return
+		}
+
+		// Load public key
```

Comment:
Can you try to move the keyStore loading before the return line 39, so we do that only once, not at every request.